src/box/errcode.h

```diff
@@ -141,287 +141,313 @@ struct errcode_record {
 	_(ER_ELECTION_DISABLED, 224,		"Elections were turned off")\
 	_(ER_TXN_ROLLBACK, 225,			"Transaction was rolled back") \
-	_(ER_NOT_LEADER, 226,			"The instance is not a leader. New leader is %u")\
+	_(ER_NOT_LEADER, 226,			"The instance is not a leader. New leader is %u", "leader_id", UINT)\
 	_(ER_SYNC_QUEUE_UNCLAIMED, 227,		"The synchronous transaction queue doesn't belong to any instance")\
-	_(ER_SYNC_QUEUE_FOREIGN, 228,		"The synchronous transaction queue belongs to other instance with id %u")\
-	_(ER_UNABLE_TO_PROCESS_IN_STREAM, 229,	"Unable to process %s request in stream") \
-	_(ER_UNABLE_TO_PROCESS_OUT_OF_STREAM, 230, "Unable to process %s request out of stream") \
+	_(ER_SYNC_QUEUE_FOREIGN, 228,		"The synchronous transaction queue belongs to other instance with id %u", "instance_id", UINT)\
+	_(ER_UNABLE_TO_PROCESS_IN_STREAM, 229,	"Unable to process %s request in stream", "request", STRING) \
+	_(ER_UNABLE_TO_PROCESS_OUT_OF_STREAM, 230, "Unable to process %s request out of stream", "request", STRING) \
 	_(ER_TRANSACTION_TIMEOUT, 231,		"Transaction has been aborted by timeout") \
 	_(ER_ACTIVE_TIMER, 232,			"Operation is not permitted if timer is already running") \
 	_(ER_TUPLE_FIELD_COUNT_LIMIT, 233,	"Tuple field count limit reached: see box.schema.FIELD_MAX") \
-	_(ER_CREATE_CONSTRAINT, 234,		"Failed to create constraint '%s' in space '%s': %s") \
-	_(ER_FIELD_CONSTRAINT_FAILED, 235,	"Check constraint '%s' failed for field '%s'") \
-	_(ER_TUPLE_CONSTRAINT_FAILED, 236,	"Check constraint '%s' failed for a tuple") \
-	_(ER_CREATE_FOREIGN_KEY, 237,		"Failed to create foreign key '%s' in space '%s': %s") \
-	_(ER_FOREIGN_KEY_INTEGRITY, 238,	"Foreign key '%s' integrity check failed: %s") \
-	_(ER_FIELD_FOREIGN_KEY_FAILED, 239,	"Foreign key constraint '%s' failed for field '%s': %s") \
-	_(ER_COMPLEX_FOREIGN_KEY_FAILED, 240,	"Foreign key constraint '%s' failed: %s") \
-	_(ER_WRONG_SPACE_UPGRADE_OPTIONS, 241,	"Wrong space upgrade options: %s") \
-	_(ER_NO_ELECTION_QUORUM, 242,		"Not enough peers connected to start elections: %d out of minimal required %d")\
+	_(ER_CREATE_CONSTRAINT, 234,		"Failed to create constraint '%s' in space '%s': %s", "constraint", STRING, "space", STRING, "details", STRING) \
+	_(ER_FIELD_CONSTRAINT_FAILED, 235,	"Check constraint '%s' failed for field '%s'", "constraint", STRING, "field", STRING, "field_id", UINT) \
+	_(ER_TUPLE_CONSTRAINT_FAILED, 236,	"Check constraint '%s' failed for a tuple", "constraint", STRING) \
+	_(ER_CREATE_FOREIGN_KEY, 237,		"Failed to create foreign key '%s' in space '%s': %s", "foreing_key", STRING, "space", STRING, "details", STRING) \
+	_(ER_FOREIGN_KEY_INTEGRITY, 238,	"Foreign key '%s' integrity check failed: %s", "foreing_key", STRING, "details", STRING) \
+	_(ER_FIELD_FOREIGN_KEY_FAILED, 239,	"Foreign key constraint '%s' failed for field '%s': %s", \
+						"constraint", STRING, "field", STRING, "details", STRING, "field_id", UINT) \
```

Comment:
Let's use "constraint" for all foreign key constraint names.

Reply:
Done.

```diff
+       _(ER_CREATE_FOREIGN_KEY, 237,           "Failed to create foreign key '%s' in space '%s': %s", "constraint", STRING, "space", STRING, "details", STRING) \
+       _(ER_FOREIGN_KEY_INTEGRITY, 238,        "Foreign key '%s' integrity check failed: %s", "constraint", STRING, "details", STRING) \
```